### ISSUE TITLE
Tutorial Code Bug for neural_net and gaussian_process Learners

### ISSUE THREAD

**Issue by matth2peters** [michaelhush/M-LOOP#122]
Hi, I've been using M-LOOP, but I've had an issue where the learner archive is not being created exclusively for the neural_net learner type. It works fine for the other learners. To distill the issue, I notice that the same thing happens in the less-complicated tutorial code on the M-LOOP website. I am using M-LOOP 3.2.1

**To Reproduce**
Execute the following code. (It's the same as the tutorial code on the website, but I have specified the controller_type='neural_net')
```

# Imports for python 2 compatibility
from __future__ import absolute_import, division, print_function

__metaclass__ = type

# Imports for M-LOOP
import mloop.interfaces as mli
import mloop.controllers as mlc
import mloop.visualizations as mlv

# Other imports
import numpy as np
import time


# Declare your custom class that inherits from the Interface class
class CustomInterface(mli.Interface):

    # Initialization of the interface, including this method is optional
    def __init__(self):
        # You must include the super command to call the parent class, Interface, constructor
        super(CustomInterface, self).__init__()

        # Attributes of the interface can be added here
        # If you want to precalculate any variables etc. this is the place to do it
        # In this example we will just define the location of the minimum
        self.minimum_params = np.array([0, 0.1, -0.1])

    # You must include the get_next_cost_dict method in your class
    # this method is called whenever M-LOOP wants to run an experiment
    def get_next_cost_dict(self, params_dict):
        # Get parameters from the provided dictionary
        params = params_dict['params']

        # Here you can include the code to run your experiment given a particular set of parameters
        # In this example we will just evaluate a sum of sinc functions
        cost = -np.sum(np.sinc(params - self.minimum_params))
        # There is no uncertainty in our result
        uncer = 0
        # The evaluation will always be a success
        bad = False
        # Add a small time delay to mimic a real experiment
        time.sleep(.01)

        # The cost, uncertainty and bad boolean must all be returned as a dictionary
        # You can include other variables you want to record as well if you want
        cost_dict = {'cost': cost, 'uncer': uncer, 'bad': bad}
        return cost_dict


def main():
    # M-LOOP can be run with three commands

    # First create your interface
    interface = CustomInterface()
    # Next create the controller. Provide it with your interface and any options you want to set
    controller = mlc.create_controller(interface,
                                       controller_type='neural_net',
                                       max_num_runs=1000,
                                       target_cost=-2.99,
                                       num_params=3,
                                       min_boundary=[-2, -2, -2],
                                       max_boundary=[2, 2, 2])
    # To run M-LOOP and find the optimal parameters just use the controller method optimize
    controller.optimize()

    # The results of the optimization will be saved to files and can also be accessed as attributes of the controller.
    print('Best parameters found:')
    print(controller.best_params)

    # You can also run the default sets of visualizations for the controller with one command
    mlv.show_all_default_visualizations(controller)


# Ensures main is run when this code is run as a script
if __name__ == '__main__':
    main()

```

The error I get from running this is

```
Traceback (most recent call last):
  File "C:/Users/matth/PycharmProjects/pythonProject/mloop_quick_test.py", line 77, in <module>
    main()
  File "C:/Users/matth/PycharmProjects/pythonProject/mloop_quick_test.py", line 72, in main
    mlv.show_all_default_visualizations(controller)
  File "C:\Users\matth\labscript-suite\Python_38\lib\site-packages\mloop\visualizations.py", line 91, in show_all_default_visualizations
    create_learner_visualizations(
  File "C:\Users\matth\labscript-suite\Python_38\lib\site-packages\mloop\visualizations.py", line 258, in create_learner_visualizations
    visualizer = create_learner_visualizer_from_archive(
  File "C:\Users\matth\labscript-suite\Python_38\lib\site-packages\mloop\visualizations.py", line 198, in create_learner_visualizer_from_archive
    controller_type = mlu.get_controller_type_from_learner_archive(filename)
  File "C:\Users\matth\labscript-suite\Python_38\lib\site-packages\mloop\utilities.py", line 274, in get_controller_type_from_learner_archive
    learner_dict = get_dict_from_file(learner_filename, file_type)
  File "C:\Users\matth\labscript-suite\Python_38\lib\site-packages\mloop\utilities.py", line 222, in get_dict_from_file
    dictionary = txt_file_to_dict(filename)
  File "C:\Users\matth\labscript-suite\Python_38\lib\site-packages\mloop\utilities.py", line 158, in txt_file_to_dict
    with open(filename,'r') as in_file:
FileNotFoundError: [Errno 2] No such file or directory: './M-LOOP_archives/learner_archive_2022-01-05_13-35.txt'

Process finished with exit code 1
```
because the learner archive file was not created. I do not get this error with the GP or DE learners. Could someone suggest a fix for this? Is there something that I misunderstood? Thanks!

**Comment by matth2peters** [comment]
Seems like this is resolved by using the version on this github, though I wasn't able to find the commit that fixed it

**Comment by zakv** [comment]
There's a possibility that this is due to a race condition.

Most of the learners save their learner archives to disk as they collect data throughout the optimization. If I recall correctly, the neural net learner waits until the end of the optimization then creates the learner archive and writes all of the data then. It's possible that somehow the plotting code is being called before the learner finishes writing the learner archive to disk after its last iteration. That wouldn't cause this error for most learners (since the file should already exist because it's been created/updated as the optimization proceeds) but would lead to a `FileNotFoundError` for the neural net learner since it doesn't create the file until the very end.

If this is the case then this error could occur randomly based on which occurs first, the saving or the plotting. They run in different processes so the order may depend on how the OS schedules when each process is run. I think that M-LOOP has some built-in inter-process flow control to prevent this issue, though I could be mistaken. Furthermore, I don't recall any recent PRs that would have changed this behavior. Please re-open this issue if you observe this error again with the most recent M-LOOP version and then I'll do some digging!

**Comment by matth2peters** [comment]
That's an interesting idea! Would that also somehow prevent it from saving the learner archive at all? As far as I can tell, the learner archive file for the NN was just never created. I couldn't run the visualizations even in a separate script because there was no learner file to load in.

**Comment by zakv** [comment]
I'm not totally sure to be honest. The `FileNotFoundError` causes the main Python process to exit, but I don't know how the other processes behave when that happens. It could be that they get killed when the main process ends, in which case the learner's process could be stopped before it reaches the point where it creates the file. I'm also not sure if that behavior depends on whether the code was run from the command line with `python script_name.py` or from an interactive Python interpreter with `import script_name.main; main()`. I'd have to do some googling/testing to find out.

**Comment by zakv** [comment]
Oh actually I may have partially figured out what the issue is here. In your example, there is only 10 ms of sleep in `get_next_cost_dict()` and `no_delay` takes its default value of `True`. When `no_delay` is `True`, then M-LOOP uses the training algorithm to suggest points if the neural net isn't ready yet. What can happen then if `get_next_cost_dict()` runs too quickly is that the entire optimization can finish before the neural nets are done being created and trained even once. That can cause errors and prevent the learner archive from being created.

If you run your example and look further up the log, do you see an error similar to the following which precedes the error you posted?
```python
Process NeuralNetLearner-1:
Traceback (most recent call last):
  File ".../lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File ".../M-LOOP/mloop/learners.py", line 2406, in run
    self._fit_neural_net(0)
  File ".../M-LOOP/mloop/learners.py", line 2271, in _fit_neural_net
    self.scaled_costs = self.cost_scaler.transform(self.all_costs[:,np.newaxis])[:,0]
AttributeError: 'NoneType' object has no attribute 'transform'
```

Furthermore, does increasing the sleep duration work around this issue? Something like 100 ms or 1 second should be enough. Similarly does passing `no_delay=False` to `mlc.create_controller()` also resolve the issue even when the sleep time is 10 ms or less?

If this is indeed the problem then I'm not 100% sure what to do about it. It seems like it would be tedious to fix/maintain the behavior when the entire optimization finishes before the learner is used, and it's only an issue in artificial examples where M-LOOP isn't very useful anyway. It is annoying though that it's easy to stumble into this issue when playing with the tutorial.

**Comment by matth2peters** [comment]
Unfortunately that doesn't solve the problem for me. I uninstalled the github repo and installed the pip version, switched no_delay=False, and the following is my output

`INFO     M-LOOP version 3.2.1
INFO     Optimization started.
INFO     Run:1 (training)
INFO     params [ 1.91020925 -1.21518481 -1.80408566]
INFO     cost 0.39844602620830727 +/- 0.0
INFO     Run:2 (training)
INFO     params [ 1.27599219  0.41857295 -0.53654925]
INFO     cost -1.3657380501418717 +/- 0.0
INFO     Run:3 (training)
INFO     params [1.20508863 0.96623763 0.38706189]
INFO     cost -0.6442510130695122 +/- 0.0
INFO     Run:4 (training)
INFO     params [-1.27633971  1.21637609 -1.82334587]
INFO     cost 0.4333293249168084 +/- 0.0
INFO     Run:5 (training)
INFO     params [-0.28341392 -1.85255862  1.98095372]
INFO     cost -0.8872897184211156 +/- 0.0
INFO     Run:6 (training)
INFO     params [1.08845327 1.87515617 0.47215754]
INFO     cost -0.34548794284440437 +/- 0.0
INFO     Run:7 (training)
INFO     params [1.03421047 0.24861303 0.6907357 ]
INFO     cost -1.1770383072774124 +/- 0.0
INFO     Run:8 (training)
INFO     params [-1.4750347   1.24374809 -0.49943103]
INFO     cost -0.42087554143337025 +/- 0.0
INFO     Run:9 (training)
INFO     params [-1.83087374  1.7612085  -1.73461792]
INFO     cost 0.4332237155244246 +/- 0.0
INFO     Run:10 (training)
INFO     params [-0.20791115 -1.51057537  1.88219979]
2022-01-10 09:40:43.478936: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2022-01-10 09:40:43.479144: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO     cost -0.7355886138840281 +/- 0.0
INFO     Run:11 (training)
INFO     params [-1.07108615  1.53622777  1.58422727]
INFO     cost 0.44122853033847825 +/- 0.0
INFO     Run:12 (machine learner)
2022-01-10 09:40:46.478963: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'nvcuda.dll'; dlerror: nvcuda.dll not found
2022-01-10 09:40:46.479185: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
2022-01-10 09:40:46.483561: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:169] retrieving CUDA diagnostic information for host: Roy87
2022-01-10 09:40:46.483801: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:176] hostname: Roy87
2022-01-10 09:40:46.484271: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
INFO     params [ 1.55686728 -0.06239085  0.1193772 ]
INFO     cost -1.678675885531152 +/- 0.0
INFO     Run:13 (machine learner)
INFO     params [ 1.76949942 -0.1744618   0.2741114 ]
INFO     cost -1.5466043550878976 +/- 0.0
INFO     Run:14 (machine learner)
INFO     params [ 2.         -0.71152392  1.82313951]
INFO     cost -0.17933545101813775 +/- 0.0
INFO     Run:15 (trainer)
INFO     params [-0.60784479 -0.70616437  1.90814298]
INFO     cost -0.7238097088425005 +/- 0.0
INFO     Run:16 (machine learner)
INFO     params [ 1.37500842 -0.10736968  0.01600203]
INFO     cost -1.694884350636746 +/- 0.0
INFO     Run:17 (machine learner)
INFO     params [ 1.36381642 -0.12345276 -0.05913406]
INFO     cost -1.7047622714029922 +/- 0.0
INFO     Run:18 (machine learner)
INFO     params [ 1.55700864 -0.31180598 -0.0638154 ]
INFO     cost -1.5401617942414798 +/- 0.0
INFO     Run:19 (trainer)
INFO     params [-1.07014245  1.974179   -0.22427455]
INFO     cost -0.844373177740563 +/- 0.0
INFO     Run:20 (machine learner)
INFO     params [ 1.41462067 -0.1638394  -0.1205094 ]
INFO     cost -1.6717037854438168 +/- 0.0
INFO     Run:21 (machine learner)
INFO     params [ 1.4854147   0.03090958 -0.03332338]
INFO     cost -1.7708041009102031 +/- 0.0
INFO     Run:22 (machine learner)
INFO     params [ 1.31737478 -0.0103874   0.06608336]
INFO     cost -1.732380816650753 +/- 0.0
INFO     Run:23 (trainer)
INFO     params [ 1.50278907  0.7307988  -1.18614841]
INFO     cost -0.17245203936208603 +/- 0.0
INFO     Run:24 (machine learner)
INFO     params [1.39781273 0.06124771 0.0089463 ]
INFO     cost -1.7620354808098866 +/- 0.0
INFO     Run:25 (machine learner)
INFO     params [ 1.55226647  0.0430778  -0.06292106]
INFO     cost -1.7901155217787006 +/- 0.0
INFO     Run:26 (machine learner)
INFO     params [ 1.44798327  0.11786954 -0.00817487]
INFO     cost -1.7687614717937423 +/- 0.0
INFO     Run:27 (trainer)
INFO     params [-0.71875476  1.02433309  0.95292944]
INFO     cost -0.3733884140166288 +/- 0.0
INFO     Run:28 (machine learner)
INFO     params [ 1.49677674  0.15018033 -0.04254535]
INFO     cost -1.7777892746706245 +/- 0.0
INFO     Run:29 (machine learner)
INFO     params [ 1.70333059  0.15337619 -0.17158251]
INFO     cost -1.8368852244673302 +/- 0.0
INFO     Run:30 (machine learner)
INFO     params [ 1.53386436  0.16365247 -0.03110568]
INFO     cost -1.779211109083656 +/- 0.0
INFO     Run:31 (trainer)
INFO     params [1.10884279 1.98492569 0.64632628]
INFO     cost -0.14905731955633383 +/- 0.0
INFO     Run:32 (machine learner)
INFO     params [ 1.68017954  0.15690107 -0.11703241]
INFO     cost -1.8343049391376525 +/- 0.0
INFO     Run:33 (machine learner)
INFO     params [ 2.          0.2265431  -0.28457646]
INFO     cost -1.918761047553275 +/- 0.0
INFO     Run:34 (machine learner)
INFO     params [ 1.77817221  0.1789742  -0.06770768]
INFO     cost -1.873162618185778 +/- 0.0
INFO     Run:35 (trainer)
INFO     params [ 0.70745854 -0.40844171 -0.94535932]
INFO     cost -1.1593556595293975 +/- 0.0
INFO     Run:36 (machine learner)
INFO     params [ 2.          0.24947902 -0.15530444]
INFO     cost -1.958625197911424 +/- 0.0
INFO     Run:37 (machine learner)
INFO     params [ 2.          0.24568087 -0.31439428]
INFO     cost -1.8915408830340859 +/- 0.0
INFO     Run:38 (machine learner)
INFO     params [ 2.          0.17244614 -0.0879144 ]
INFO     cost -1.9911487444381448 +/- 0.0
INFO     Run:39 (trainer)
INFO     params [-1.23975904 -1.28143768  0.86223312]
INFO     cost 0.3510788945069205 +/- 0.0
INFO     Run:40 (machine learner)
INFO     params [ 2.          0.23578357 -0.11775871]
INFO     cost -1.969428116364985 +/- 0.0
INFO     Run:41 (machine learner)
INFO     params [ 2.          0.23315853 -0.22145478]
INFO     cost -1.9469986704199447 +/- 0.0
INFO     Run:42 (machine learner)
INFO     params [ 2.          0.21630275 -0.12952845]
INFO     cost -1.9764644757167353 +/- 0.0
INFO     Run:43 (trainer)
INFO     params [-0.23086601  0.33110955 -1.11430387]
INFO     cost -1.8149346670308375 +/- 0.0
INFO     Run:44 (machine learner)
INFO     params [ 2.          0.2215626  -0.06808278]
INFO     cost -1.9741938247247561 +/- 0.0
INFO     Run:45 (machine learner)
INFO     params [ 2.          0.22438017 -0.11224418]
INFO     cost -1.974499152056107 +/- 0.0
INFO     Run:46 (machine learner)
INFO     params [ 2.          0.21087985 -0.11727272]
INFO     cost -1.979408271045624 +/- 0.0
INFO     Run:47 (trainer)
INFO     params [-1.85274624  1.07211272 -1.04002494]
INFO     cost -0.015401367804075368 +/- 0.0
INFO     Run:48 (machine learner)
INFO     params [ 2.          0.18476997 -0.09350409]
INFO     cost -1.988152026556612 +/- 0.0
INFO     Run:49 (machine learner)
INFO     params [ 2.          0.18196004 -0.11335355]
INFO     cost -1.9886935180939633 +/- 0.0
INFO     Run:50 (machine learner)
INFO     params [ 2.          0.18797932 -0.12286319]
INFO     cost -1.9864565344616945 +/- 0.0
INFO     Run:51 (trainer)
INFO     params [-0.35109393  0.65898068  0.81621381]
INFO     cost -1.4593122196318613 +/- 0.0
INFO     Run:52 (machine learner)
INFO     params [ 2.          0.18954482 -0.09230098]
INFO     cost -1.9867650566323172 +/- 0.0
INFO     Run:53 (machine learner)
INFO     params [ 2.          0.20165009 -0.11811625]
INFO     cost -1.9825500014495998 +/- 0.0
INFO     Run:54 (machine learner)
INFO     params [ 2.          0.1755186  -0.12949136]
INFO     cost -1.9892151580444921 +/- 0.0
INFO     Run:55 (trainer)
INFO     params [-0.27047568 -0.55095056  0.09084249]
INFO     cost -2.26012010356675 +/- 0.0
INFO     Run:56 (machine learner)
INFO     params [-0.00350844  0.44042271 -0.56931388]
INFO     cost -2.4950576743549107 +/- 0.0
INFO     Run:57 (machine learner)
INFO     params [ 2.          0.16524356 -0.09442939]
INFO     cost -1.9929616225288684 +/- 0.0
INFO     Run:58 (machine learner)
INFO     params [0.05229845 0.17871127 0.03148658]
INFO     cost -2.95714978803307 +/- 0.0
INFO     Run:59 (trainer)
INFO     params [-1.49867834  0.67278952  1.0495786 ]
INFO     cost -0.20348026758285986 +/- 0.0
INFO     Run:60 (machine learner)
INFO     params [0.01894473 0.25069891 0.08316757]
INFO     cost -2.908187864425565 +/- 0.0
INFO     Run:61 (machine learner)
INFO     params [-0.16895263  0.14334178  0.06730622]
INFO     cost -2.9052030991272852 +/- 0.0
INFO     Run:62 (machine learner)
INFO     params [ 0.02702558  0.04900553 -0.00718017]
INFO     cost -2.9804151108369377 +/- 0.0
INFO     Run:63 (trainer)
INFO     params [-0.90962619 -0.77554457 -0.35501173]
INFO     cost -1.1329846760517965 +/- 0.0
INFO     Run:64 (machine learner)
INFO     params [0.04078726 0.18657835 0.0567039 ]
INFO     cost -2.945074628380934 +/- 0.0
INFO     Run:65 (machine learner)
INFO     params [0.00921254 0.08338385 0.01043212]
INFO     cost -2.979466292708233 +/- 0.0
INFO     Run:66 (machine learner)
INFO     params [ 0.00192515  0.0591716  -0.01252191]
INFO     cost -2.984713851185348 +/- 0.0
INFO     Run:67 (trainer)
INFO     params [-0.36434209  0.45228053  0.91736509]
INFO     cost -1.5864529528530398 +/- 0.0
INFO     Run:68 (machine learner)
INFO     params [0.02357219 0.19540522 0.07677208]
INFO     cost -2.933566241051533 +/- 0.0
INFO     Run:69 (machine learner)
INFO     params [-0.05051618  0.07592448  0.10524717]
INFO     cost -2.926985635799519 +/- 0.0
INFO     Run:70 (machine learner)
INFO     params [ 0.00297664  0.07434434 -0.01597563]
INFO     cost -2.9873300653096195 +/- 0.0
INFO     Run:71 (trainer)
INFO     params [ 0.29183256 -0.17958069 -1.14877289]
INFO     cost -1.6956481207531398 +/- 0.0
INFO     Run:72 (machine learner)
INFO     params [ 0.03488832  0.0793008  -0.01870937]
INFO     cost -2.9864597559315413 +/- 0.0
INFO     Run:73 (machine learner)
INFO     params [0.03780249 0.05262496 0.00896194]
INFO     cost -2.974547495751618 +/- 0.0
INFO     Run:74 (machine learner)
INFO     params [ 0.00614416  0.07014731 -0.02817281]
INFO     cost -2.9900077384146058 +/- 0.0
INFO     Controller finished. Closing down M-LOOP. Please wait a moment...
Process NeuralNetLearner-1:
Traceback (most recent call last):
  File "C:\Python38\lib\multiprocessing\process.py", line 315, in _bootstrap
    self.run()
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\learners.py", line 2401, in run
    self.find_global_minima(0)
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\learners.py", line 2443, in find_global_minima
    self.predicted_best_cost = float(self.cost_scaler.inverse_transform([self.predicted_best_scaled_cost]))
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\sklearn\preprocessing\_data.py", line 1016, in inverse_transform
    X = check_array(
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\sklearn\utils\validation.py", line 769, in check_array
    raise ValueError(
ValueError: Expected 2D array, got 1D array instead:
array=[-2.98931939].
Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.
INFO     Machine learning learner did not provide best and/or all minima.
INFO     Optimization ended because:-
INFO     Target cost reached.
INFO     Results:-
INFO     Best parameters found:[ 0.00614416  0.07014731 -0.02817281]
INFO     Best cost returned:-2.9900077384146058 +/- 0.0
INFO     Best run number:74
INFO     M-LOOP Done.
Best parameters found:
[ 0.00614416  0.07014731 -0.02817281]
Traceback (most recent call last):
  File "C:/Users/matth/PycharmProjects/pythonProject/mloop_quick_testv2.py", line 79, in <module>
    main()
  File "C:/Users/matth/PycharmProjects/pythonProject/mloop_quick_testv2.py", line 74, in main
    mlv.show_all_default_visualizations(controller)
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\visualizations.py", line 91, in show_all_default_visualizations
    create_learner_visualizations(
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\visualizations.py", line 258, in create_learner_visualizations
    visualizer = create_learner_visualizer_from_archive(
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\visualizations.py", line 198, in create_learner_visualizer_from_archive
    controller_type = mlu.get_controller_type_from_learner_archive(filename)
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\utilities.py", line 274, in get_controller_type_from_learner_archive
    learner_dict = get_dict_from_file(learner_filename, file_type)
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\utilities.py", line 222, in get_dict_from_file
    dictionary = txt_file_to_dict(filename)
  File "C:\Users\matth\PycharmProjects\pythonProject\venv2\lib\site-packages\mloop\utilities.py", line 158, in txt_file_to_dict
    with open(filename,'r') as in_file:
FileNotFoundError: [Errno 2] No such file or directory: './M-LOOP_archives/learner_archive_2022-01-10_09-40.txt'
`

**Comment by zakv** [comment]
Could you try the same tests but using the most up-to-date Github version? You may be running into issue #120 which was fixed by #121, though that patch isn't in a release version of M-LOOP yet. If that doesn't work please post the whole output log and I can dig deeper.

**Comment by matth2peters** [comment]
Oh, I misunderstood you and I was unclear with my previous comments. I was able to resolve this issue by installing from the github repo instead of the release version, so it does work correctly with the most recent PRs. Sorry about that confusion!

**Comment by zakv** [comment]
Oh gotcha. Strangely I was able to produce the same error in a different way even with the most recent version from Github. I bet the error you were seeing was fixed by #121, so I'll go ahead and close this issue again. Feel free to reopen it if I'm mistaken.
